### ISSUE TITLE
Where is the trained model?

### ISSUE THREAD

**Issue by PanZiwei** [yaozhong/methBERT#2]
Hi,
Thanks for the methBERT tool. You mentioned that "We provided independent trained models on each 5mC and 6mA datasets of different motifs and methyltransferases in the ./trained_model fold.", but I didn't find the folder.  Any chance to provide the pre-trained BERT model for direct usage? Thanks!

**Comment by yaozhong** [comment]
HI, PanZiwei,

Thanks for contacting us. 
We have uploaded the 39 models reported in our BIBM21 paper.
All model can be loaded through the basic loading approach of PyTorch, like

state_dict = torch.load(model_path)
net.load_state_dict(state_dict)	

For reference, we provide tree structure of files as follows:

trained_model/
├── Simpson
│   └── random_split_balanced
│       ├── BERT_plus_W21_E50_simpson_human-x_basic_lr-1e-4-128.pth
│       ├── BERT_W21_E50_simpson_human-x_basic_lr-1e-4-128.pth
│       └── biRNN_basic_W21_E50_simpson_human-x_basic_lr-1e-4-128.pth
└── Stoiber
    ├── random_split_balanced
    │   ├── R9_5mC
    │   │   ├── M_Hhal_gCgc
    │   │   │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_Hhal_gCgc_basic_lr-1e-4-128.pth
    │   │   │   ├── BERT_W21_E50_stoiber_ecoli-M_Hhal_gCgc_basic_lr-1e-4-128.pth
    │   │   │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_Hhal_gCgc_basic_lr-1e-4-128.pth
    │   │   ├── M_Mpel_Cg
    │   │   │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_Mpel_Cg_basic_lr-1e-4-128.pth
    │   │   │   ├── BERT_W21_E50_stoiber_ecoli-M_Mpel_Cg_basic_lr-1e-4-128.pth
    │   │   │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_Mpel_Cg_basic_lr-1e-4-128.pth
    │   │   └── M_Sssl_Cg
    │   │       ├── BERT_plus_W21_E50_stoiber_ecoli-M_Sssl_Cg_basic_lr-1e-4-128.pth
    │   │       ├── BERT_W21_E50_stoiber_ecoli-M_Sssl_Cg_basic_lr-1e-4-128.pth
    │   │       └── biRNN_basic_W21_E50_stoiber_ecoli-M_Sssl_Cg_basic_lr-1e-4-128.pth
    │   └── R9_6mA
    │       ├── M_dam_gAtc
    │       │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_dam_gAtc_basic_lr-1e-4-128.pth
    │       │   ├── BERT_W21_E50_stoiber_ecoli-M_dam_gAtc_basic_lr-1e-4-128.pth
    │       │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_dam_gAtc_basic_lr-1e-4-128.pth
    │       ├── M_EcoRI_gaAttc
    │       │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_EcoRI_gaAttc_basic_lr-1e-4-128.pth
    │       │   ├── BERT_W21_E50_stoiber_ecoli-M_EcoRI_gaAttc_basic_lr-1e-4-128.pth
    │       │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_EcoRI_gaAttc_basic_lr-1e-4-128.pth
    │       └── M_TaqI_tcgA
    │           ├── BERT_plus_W21_E50_stoiber_ecoli-M_TaqI_tcgA_basic_lr-1e-4-128.pth
    │           ├── BERT_W21_E50_stoiber_ecoli-M_TaqI_tcgA_basic_lr-1e-4-128.pth
    │           └── biRNN_basic_W21_E50_stoiber_ecoli-M_TaqI_tcgA_basic_lr-1e-4-128.pth
    └── region_based
        ├── R9_5mC
        │   ├── M_Hhal_gCgc
        │   │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_Hhal_gCgc_basic_lr-1e-4-128_balanced_region.pth
        │   │   ├── BERT_W21_E50_stoiber_ecoli-M_Hhal_gCgc_basic_lr-1e-4-128_balanced_region.pth
        │   │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_Hhal_gCgc_basic_lr-1e-4-128_balanced_region.pth
        │   ├── M_Mpel_Cg
        │   │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_Mpel_Cg_basic_lr-1e-4-128_balanced_region.pth
        │   │   ├── BERT_W21_E50_stoiber_ecoli-M_Mpel_Cg_basic_lr-1e-4-128_balanced_region.pth
        │   │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_Mpel_Cg_basic_lr-1e-4-128_balanced_region.pth
        │   └── M_Sssl_Cg
        │       ├── BERT_plus_W21_E50_stoiber_ecoli-M_Sssl_Cg_basic_lr-1e-4-128_balanced_region.pth
        │       ├── BERT_W21_E50_stoiber_ecoli-M_Sssl_Cg_basic_lr-1e-4-128_balanced_region.pth
        │       └── biRNN_basic_W21_E50_stoiber_ecoli-M_Sssl_Cg_basic_lr-1e-4-128_balanced_region.pth
        └── R9_6mA
            ├── M_dam_gAtc
            │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_dam_gAtc_basic_lr-1e-4-128_balanced_region.pth
            │   ├── BERT_W21_E50_stoiber_ecoli-M_dam_gAtc_basic_lr-1e-4-128_balanced_region.pth
            │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_dam_gAtc_basic_lr-1e-4-128_balanced_region.pth
            ├── M_EcoRI_gaAttc
            │   ├── BERT_plus_W21_E50_stoiber_ecoli-M_EcoRI_gaAttc_basic_lr-1e-4-128_balanced_region.pth
            │   ├── BERT_W21_E50_stoiber_ecoli-M_EcoRI_gaAttc_basic_lr-1e-4-128_balanced_region.pth
            │   └── biRNN_basic_W21_E50_stoiber_ecoli-M_EcoRI_gaAttc_basic_lr-1e-4-128_balanced_region.pth
            └── M_TaqI_tcgA
                ├── BERT_plus_W21_E50_stoiber_ecoli-M_TaqI_tcgA_basic_lr-1e-4-128_balanced_region.pth
                ├── BERT_W21_E50_stoiber_ecoli-M_TaqI_tcgA_basic_lr-1e-4-128_balanced_region.pth
                └── biRNN_basic_W21_E50_stoiber_ecoli-M_TaqI_tcgA_basic_lr-1e-4-128_balanced_region.pth

Thanks!

**Comment by PanZiwei** [comment]
Hi Yaozhong,
Congratulations on your BIBM21 paper! 2 Following questions:
1. Will the BIBM21 paper be available later? The online version is not available yet (https://ieeexplore.ieee.org/document/9669841)...Or it is the same version as the bioxiv one?
2. Which models can be used on the human dataset? 

Thank you so much for your help!

**Comment by yaozhong** [comment]
Hi, PanZiwei,

Thanks for your inquiry.

1. The paper is available at https://ieeexplore.ieee.org/document/9669841
    This manuscript version **is different** from the first version in bioAxiv. The models are retrained using balanced positive and negative training samples, while the model architectures are still the same.

2. In the paper, we evaluate model performances based on publicly available benchmark datasets using R9 flowcells. As the inputs of the deep learning models are normalized statistical measures of signals, it is possible to use the provided models trained on Simpson's human dataset, such as *_W21_E50_simpson_human-x_basic_lr-1e-4-128.pth. But its generalization to the current R9.4/R10 dataset is not systematically evaluated. 
If possible, we suggest directly retraining the model on the dataset that is similar to your sequencing setting.
For example, your can use R9.4 NA12878  or other available datasets.

Currently, we are also working on an R9.4 human cell line dataset. We hope it will assist your research very soon.
